lib/media/streaming_engine.js

```diff
@@ -329,14 +333,51 @@ shaka.media.StreamingEngine = class {
    *   reset the timestamp offset to ensure the new track segments are correctly
    *   placed on the timeline.
    */
-  switchVariant(
-      variant, clearBuffer = false, safeMargin = 0, force = false,
-      adaptation = false) {
+  async switchVariant(variant, clearBuffer = false,
```

Comment:
Also, for reference, I spent a while considering this particular change.
The fact that this method was made async, but none of the callsites for it were made to `await` the result, made me kind of nervous.
But all of the asynchronous operations inside this method are just methods on the MediaSourceEngine, which uses a queue to protect against race conditions and other concurrency errors. So it should probably be safe.
Also, actually awaiting this every time would require changing the public interface of the player, which would make this a good bit more complicated to put into a feature release.